lib/user/user-session.js

```diff
@@ -0,0 +1,53 @@
+var express = require('express')
+  , config = require('./config-user')
+  , Delegate = require('./user-delegate');
+
+function initRouter(delegate) {
+  var router = express.Router();
+
+  router.route('/auth').post(function(req, res, next) {
+    var params = req && req.body && req.body.params;
+    delegate.auth(params).then(function(data) {
+      res.json(data);
+    }, function(error) {
+      next(error);
+    });
+  });
+
+  router.route('/verifysession').post(function(req, res, next) {
+    delegate.verifysession().then(function(data) {
+      res.json(data);
+    }, function(error) {
+      next(error);
+    });
+  });
+
+  router.route('/revokesession').post(function(req, res, next) {
+    delegate.revokesession().then(function(data) {
+      res.json(data);
+    }, function(error) {
+      next(error);
+    });
+  });
+
+  return router;
+}
+
+module.exports = function(mediator, app, guid) {
+  var self = this;
+  self.mediator - mediator;
+
+  var delegate = new Delegate(guid);
+  var router = initRouter(delegate);
+  app.use(config.authpolicyPath, router);
+
+  mediator.subscribe('wfm:session:validate', function(sessId) {
+    return delegate.verifysession(sessId)
+      .then(function() {
+        self.mediator.publish('done:' + 'wfm:session:validate');
```

Comment:
It needs to publish the validation response for the consumer.

self.mediator.publish('done:' + 'wfm:session:validate', validationResponse);

See https://github.com/feedhenry-raincatcher/raincatcher-user/pull/13/files#diff-672c3d0fb13ccf11671e05c394876da3R49